src/views/hierachicalPackageRootNode.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { INodeData, NodeKind } from "../java/nodeData";
+import { PackageTreeNode } from "../java/packageTreeNode";
+import { Settings } from "../settings";
+import { DataNode } from "./dataNode";
+import { ExplorerNode } from "./explorerNode";
+import { FileNode } from "./fileNode";
+import { FolderNode } from "./folderNode";
+import { HierachicalPackageRootSubNode } from "./hierachicalPackageRootSubNode";
+import { PackageRootNode } from "./packageRootNode";
+import { ProjectNode } from "./projectNode";
+import { TypeRootNode } from "./typeRootNode";
+
+export class HierachicalPackageRootNode extends PackageRootNode {
+
+    public static async convertPaths(paths: INodeData[]): Promise<INodeData[]> {
+        const index = paths.findIndex((nodeData) => nodeData.kind === NodeKind.PackageRoot);
+        const projectNodeData = paths.find((nodeData) => nodeData.kind === NodeKind.Project);
```

Comment:
All these operations are model oriented. You can compute the tree without creating a view. 

For example, you can make the packagetree as a the model and build the hierarchy model outof the the view.